core/state_transition.go

```diff
@@ -227,12 +288,36 @@ func (st *StateTransition) TransitionDb() (ret []byte, usedGas uint64, failed bo
 		}
 	}
 	st.refundGas()
+	if st.isEIP1559 {
+		// block.coinbase gains (gasprice - BASEFEE) * gasused
+		coinBaseCredit := new(big.Int).Mul(new(big.Int).Sub(st.eip1559GasPrice, st.evm.BaseFee), new(big.Int).SetUint64(st.gasUsed()))
+		//  If gasprice < BASEFEE (due to the fee_cap), this means that the block.coinbase loses funds from this operation;
```

Comment:
Do we accept the block coinbase to lose funds from a transaction if gas price < BASEFEE because of fee cap ?

Reply:
Vitalik thinks we should not accept. I quote : `problem with allowing that is that a miner could choose someone else as coinbase and drain their money`
See https://discordapp.com/channels/595666850260713488/692078615269212180/703225803143643238

Reply:
Thank you for the heads up! Making the adjustment now.

Reply:
BTW could you change the `MaxGasEIP1559` to 20 million ? The EIP needs to be updated as well.
For context: https://discordapp.com/channels/595666850260713488/692078615269212180/705078808167841852

Reply:
Are you on the discord channel ? I am working on Besu implementation so that would be easier to sync our efforts if we could chat on discord ? Does that make sense ?

Reply:
Updated `MaxGasEIP1559`! Yeah I am in the discord channel, let's coordinate there.